hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ec/reconstruction/ECReconstructionCoordinator.java

```diff
@@ -323,9 +323,13 @@ public void reconstructECBlockGroup(BlockLocationInfo blockLocationInfo,
             }
             // TODO: can be submitted in parallel
             for (int i = 0; i < bufs.length; i++) {
-              CompletableFuture<ContainerProtos.ContainerCommandResponseProto>
-                  future = targetBlockStreams[i].write(bufs[i]);
-              checkFailures(targetBlockStreams[i], future);
+              if (bufs[i].limit() != 0) {
```

Comment:
While this works OK, I think a better condition would be:

```suggestion
              if (bufs[i].remaining() > 0) {
```